pwiz_tools/Skyline/Model/DocSettings/TransitionIonMobilityFiltering.cs

```diff
@@ -0,0 +1,1180 @@
+            {
+                var ionMobilityUnitsL10NString = IonMobilityUnitsL10NString(u);
+                if (string.Equals(units, ionMobilityUnitsL10NString, StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(units, u.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    result = u;
+                    return true;
+                }
+            }
+            // No match - maybe it's in a different language
+            var success = false;
+            var currentCulture = Thread.CurrentThread.CurrentUICulture;
+            try
+            {
+                foreach (var culture in new[] {@"en", @"zh-CHS", @"ja"})
+                {
+                    var tryCulture = new CultureInfo(culture);
+                    Thread.CurrentThread.CurrentUICulture = tryCulture;
+                    foreach (eIonMobilityUnits u in Enum.GetValues(typeof(eIonMobilityUnits)))
+                    {
+                        if (u != eIonMobilityUnits.none)
+                        {
+                            var ionMobilityUnitsL10NString = IonMobilityUnitsL10NString(u);
+                            if (string.Equals(units, ionMobilityUnitsL10NString, StringComparison.OrdinalIgnoreCase))
+                            {
+                                result = u;
+                                success = true;
+                                break;
+                            }
+                        }
+                    }
+                    if (success)
+                        break;
+                }
+            }
+            finally
+            {
+                Thread.CurrentThread.CurrentUICulture = currentCulture;
+            }
+
+            return success;
+        }
+
+        public static string[] KnownIonMobilityTypes
+        {
+            get
+            {
+                var result = new List<string>();
+                var currentCulture = Thread.CurrentThread.CurrentUICulture;
+                foreach (var culture in new[] { @"en", @"zh-CHS", @"ja" })
+                {
+                    var tryCulture = new CultureInfo(culture);
+                    Thread.CurrentThread.CurrentUICulture = tryCulture;
+                    foreach (eIonMobilityUnits u in Enum.GetValues(typeof(eIonMobilityUnits)))
+                    {
+                        if (u != eIonMobilityUnits.none && u!= eIonMobilityUnits.unknown)
+                            result.Add(IonMobilityUnitsL10NString(u));
+                    }
+                }
+                Thread.CurrentThread.CurrentUICulture = currentCulture;
+                return result.ToArray();
+            }
+        }
+
+        public enum SerializationElementType { TransitionChromInfo, TransitionGroupChromInfo, LegacyMeasuredIonMobility }
+
+        private double? PersistRound(double? val)
+        {
+            return val.HasValue ? Math.Round(val.Value, 6) : (double?)null;
```

Comment:
Quite so. This is some fairly old code, that got revived, our thinking may have changed since then,  I'll see about making sure that anything that roundtrips does so at full resolution.